BlindexDAO/BlinDEX
# Feature/lago 735 zap mint contract implementation

ZapMint todo list:
* Test zap mint for 0% collateral ratio
* Optimise gas usage
* Provide multi-AMM functionality (each swap can be atomic and executed on different AMM)
* use one initialized zapmint object for all tests (alternatively - move initialization to private method - partially done)
* move common variables to constants (like deadline, router)
* in case bdx swap fails:
     ** swap collateral token back to input token
     ** return input token to user
* in case mint fails:
     ** swap collateral token back to input token
     ** swap bdx token back to input token
     ** return input token to user
* implement functionality to zap mint based on 'required amount out' (more below)
* add tests for all new require statements
* add test for reentrancy
* fix all compile warnings (npm run compile)

Open questions:
* if user has any collateral/bdx should the contract allow for transferring that instead of swapping?
* instead of swapping token back in case of swap/mint fail the treasury could return lost tokens to user
